Services/DeviceMethods.cs

```diff
@@ -13,35 +13,64 @@
 
 namespace Microsoft.Azure.IoTSolutions.DeviceSimulation.Services
 {
-    public class DeviceMethods
+    public interface IDeviceMethods
     {
-        private static readonly TimeSpan retryMethodCallbackRegistration = TimeSpan.FromSeconds(10);
+        Task RegisterMethodsAsync(
+            string deviceId,
+            IDictionary<string, Script> methods,
+            Dictionary<string, object> deviceState);
+    }
 
+    public class DeviceMethods : IDeviceMethods
+    {
         private readonly Azure.Devices.Client.DeviceClient client;
         private readonly ILogger log;
         private readonly IScriptInterpreter scriptInterpreter;
-        private readonly IDictionary<string, Script> cloudToDeviceMethods;
-        private readonly Dictionary<string, object> deviceState;
-        private readonly string deviceId;
+        private IDictionary<string, Script> cloudToDeviceMethods;
+        private Dictionary<string, object> deviceState;
+        private string deviceId;
```

Comment:
should deviceid still be set in the constructor?

Reply:
I see why we could, but that would make DI harder. I'll add a check in `RegisterMethodsAsync` to make sure the ID is not overwritten, i.e. each device should have a dedicated instance of this class